### ISSUE TITLE
lineHeight style on outer Text is not applied to nested Text on iOS

### ISSUE THREAD

**Issue by alexisbronchart** [facebook/react-native#19051]
- [x] I have reviewed the [documentation](https://facebook.github.io/react-native)
- [x] I have searched [existing issues](https://github.com/facebook/react-native/issues)
- [x] I am using the [latest React Native version](https://github.com/facebook/react-native/releases)

Setting the lineHeight style like in the following code has no effect on iOS.
```
render() {
    return (
      <View>
        <Text style={{ lineHeight: 24 }}>
          <Text>
            an error occured an error occured an error occured an error occured
            an error occured an error occured an error occured an error occured
            an error occured an error occured an error occured an error occured
            an error occured an error occured an error occured an error occured
            an error occured an error occured an error occured an error occured
            an error occured an error occured
          </Text>
        </Text>
      </View>
    )
  }
```

## Environment
Environment:
  OS: macOS High Sierra 10.13.4
  Node: 9.5.0
  Yarn: 1.6.0
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.3 Build version 9E145
  Android Studio: 3.1 AI-173.4720617

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: 0.55.3 => 0.55.3

## Steps to Reproduce
Render a simple view with the code above (on iOS).

## Expected Behavior
I would expect the lines to be more spaced (like it does when I apply the style on the inner Text).

## Actual Behavior
The text renders the same as if I wouldn't set any style.

**Comment by aparedes** [comment]
This is fixed by #18340 but it has not landed in a release. Hopefully it will be added in 0.56.0

**Comment by aparedes** [comment]
It is fixed in 0.56.0
